src/sidebar/panel.js

```diff
@@ -201,14 +233,68 @@ function getThemeFromStorage() {
     }
   });
 }
+
+function getLastSyncedTime() {
+  const getting = browser.storage.local.get(['last_modified', 'credentials']);
+  getting.then(data => {
+    if (data.hasOwnProperty('credentials')) {
+      const time = new Date(data.last_modified).toLocaleTimeString();
+      enableSync.textContent = 'Synced at ' + time;
+    }
+  });
+}
+
 document.addEventListener('DOMContentLoaded', getThemeFromStorage);
+document.addEventListener('DOMContentLoaded', getLastSyncedTime);
 
 chrome.runtime.onMessage.addListener(eventData => {
+  let time;
+  let content;
   switch (eventData.action) {
+    case 'sync-authenticated':
+      chrome.runtime.sendMessage({
+          action: 'kinto-load'
+        });
+      break;
+    case 'kinto-loaded':
+      console.log("kinto-loaded content", eventData);
+      if (eventData.data !== null) {
+        const local = quill.getContents();
+        const remote = eventData.data;
+        if (!eventData.contentWasSynced) {
+          const newContent = JSON.parse(JSON.stringify(remote));
+          newContent.ops.push({ insert: '\n====== Previously was ======\n\n' });
+          content = newContent.ops.concat(local.ops);
+        } else {
+          content = remote;
+        }
+        ignoreNextTextChange = true;
+      } else {
+        browser.storage.local.remove('initialContent');
+        content = eventData.data;
+      }
+    browser.storage.local.set({ notes: content, last_modified: eventData.last_modified });
+
+      setTimeout(() => {
+        console.log('Content is', content);
+        quill.setContents(content);
+      }, 10);
+      break;
     case 'text-change':
       ignoreNextLoadEvent = true;
       loadContent();
       break;
+    case 'text-editing':
+      enableSync.textContent = 'Editing';
+      break;
+    case 'text-synced':
+      time = new Date(eventData.last_modified).toLocaleTimeString();
+      enableSync.textContent = 'Synced at ' + time;
+      break;
+    case 'text-saved':
+      time = new Date().toLocaleTimeString();
+      enableSync.textContent = 'Saved at ' + time;
+      break;
     case 'theme-changed':
       getThemeFromStorage();
```

Comment:
Nit: is this a case without a `break`? That could cause some sketchy bugs in the future. ⚠️